src/Raven.Studio/typescript/components/pages/database/tasks/ongoingTasks/editTasks/editGenAiTask/partials/fields/EditGenAiTaskNodeField.tsx

```diff
@@ -0,0 +1,76 @@
+import { FormGroup, FormSelect, FormSwitch } from "components/common/Form";
+import RichAlert from "components/common/RichAlert";
+import { clusterSelectors } from "components/common/shell/clusterSlice";
+import { useAppSelector } from "components/store";
+import { useFormContext, useWatch } from "react-hook-form";
+import { EditGenAiTaskFormData } from "../../utils/editGenAiTaskValidation";
+
+export default function EditGenAiTaskNodeField() {
+    const { control } = useFormContext<EditGenAiTaskFormData>();
+    const formValues = useWatch<EditGenAiTaskFormData>({ control });
+
+    const nodes = useAppSelector(clusterSelectors.allNodes);
+
+    const possibleMentors = nodes.filter((x) => x.type === "Member").map((x) => x.nodeTag);
```

Comment:
Imo, rename `possibleMentors` to `possibleMentorsOptions` and build mapped `[value,label]` array. It will work the same way, and it will be more readable.
```suggestion
    const possibleMentorsOptions = nodes
  .filter(node => node.type === 'Member')
  .map(node => ({ value: node.nodeTag, label: `Node ${node.nodeTag}` }));
```